woocommerce/woocommerce-gutenberg-products-block
# Only validate postcode if its required or filled

We are validating postcodes even if they're not required, in this PR, I change that so we only validate them if they're required or filled.

The filled check prevent adding invalid optional postcodes.


Fixes #9376

#### User Facing Testing

## Testing optional empty fields
1. Set country to something with an optional postcode like Vietnam
2. Try to checkout with the field empty, it should pass fine.


## Testing optional filled fields

For this one we need a turn a required country to optional, this can, add this code to make the UK's postcode optional

```php
add_filter(
	'woocommerce_get_country_locale',
	function( $locale ) {
		$locale['GB']['postcode']['required'] = false;
		return $locale;
	},
	10,
	1
);
```
- In Checkout, use an invalid postcode valid.
- Try to checkout, validation should kick in.
- Empty the field, you should checkout fine.
- 
### Changelog

> Fix bug in which optional postcodes were still validated.
